### PR TITLE
[Admin][Behat] Cover scenarios for admin password reset

### PR DESCRIPTION
**Pull request by jakubtobiasz**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | master           |
| Bug fix?        | no                                                       |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no  |
| Related tickets |                       |
| License         | MIT                                                          |

Scenarios covering `Admin password reset` feature.

### CODE REVIEW

**Review by lchrusciel — APPROVED** [review]

File: features/admin/resetting_password.feature [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    Background:
+        Given there is an administrator "sylius@example.com" identified by "sylius"
+
+    @todo
+    Scenario: Resetting an administrator password
+        When I want to reset password
+        And I specify email as "sylius@example.com"
+        And I reset it
+        Then I should be notified that email with reset instruction has been sent
+        And an email with reset token should be sent to "sylius@example.com"
```
Comment (AdamKasp) [inline features/admin/resetting_password.feature:16]: the administrator should not be aware of the "reset token", he only gets an email with a link and instructions on how to change the password
`And an email with instructions on how to reset the password should be sent to "sylius@example.com"

File: features/admin/resetting_password.feature [hunk 2]
```diff
@@ -0,0 +1,35 @@
+@administrator_login
+Feature: Resetting an administrator's password
+    In order to login to my administrator account when I forget my password
+    As an not logged in Administrator
+    I need to be able to reset my password
```
Comment (AdamKasp) [inline features/admin/resetting_password.feature:5]:
```suggestion
    I want to be able to reset my password
```

File: features/admin/resetting_password.feature [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    Background:
+        Given there is an administrator "sylius@example.com" identified by "sylius"
+
+    @todo
+    Scenario: Resetting an administrator password
```
Comment (AdamKasp) [inline features/admin/resetting_password.feature:11]: it isn't resetting the password but only sending an email with the next steps.
probably you should change this scenario name

File: features/admin/resetting_password.feature [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        When I want to reset password
+        And I specify email as "sylius@example.com"
+        And I reset it
+        Then I should be notified that email with reset instruction has been sent
+        And an email with reset token should be sent to "sylius@example.com"
+
+    @todo
+    Scenario: Changing my administrator password with a token I received
```
Comment (AdamKasp) [inline features/admin/resetting_password.feature:19]:
```suggestion
    Scenario: Changing my administrator password
```

File: features/admin/resetting_password.feature [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        When I follow link on my email to reset my password
+        And I specify my new password as "newp@ssw0rd"
+        And I confirm my new password as "newp@ssw0rd"
+        And I reset it
+        Then I should be notified that my password has been successfully changed
+        And I should be able to log in as "sylius@example.com" with "newp@ssw0rd" password
+
+    @todo
+    Scenario: Trying to change my administrator password twice with a token I received
+        Given I have already received a resetting administrator's password email
```
Comment (AdamKasp) [inline features/admin/resetting_password.feature:30]: If I see correctly we mist one step sth like `And this email has been already used`

File: features/admin/resetting_password.feature [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        And I confirm my new password as "newp@ssw0rd"
+        And I reset it
+        Then I should be notified that my password has been successfully changed
+        And I should be able to log in as "sylius@example.com" with "newp@ssw0rd" password
+
+    @todo
+    Scenario: Trying to change my administrator's password twice without sending a new password reset request
+        Given I already reset my administrator's password
+        When I try to reset my password again using the same email
+        Then I should not be able to change my password again without sending a new password reset request
```
Comment (AdamKasp) [inline features/admin/resetting_password.feature:32]:
```suggestion
        Then I should not be able to change it again without sending a new password reset request
```

**Review by AdamKasp — APPROVED** [review]

**Review by Zales0123 — APPROVED** [review]

**Pull request merged**